dingo/gw/inference/inference_pipeline.py

```diff
@@ -241,6 +244,17 @@ def prepare_log_prob(
         sampler.to_hdf5(label=low_latency_label, outdir=outdir)
     result = sampler.to_result()
     nde_settings["training"]["device"] = str(sampler.model.device)
+
+    # If there is only one parameter, and no context, we cannot use a
+    # coupling transform. In this case, we use an autoregressive transform
+    # for the density estimator
+    if len(sampler.gnpe_proxy_parameters) == 1:
+        print(
+            "Using autoregressive transform for density estimator since there is only 1 GNPE proxy."
+        )
+        nde_settings["model"]["base_transform_kwargs"][
+            "base_transform_type"
+        ] = "rq-autoregressive"
```

Comment:
I think choosing these options might be better done in `dingo_pipe`, where the `density_recovery_settings` get chosen. The `prepare_log_prob()` function should just take those settings and run them.